File: problems/Java/129. Sum Root to Leaf Numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return dfs(root, 0);
    }

    private int dfs(TreeNode n, int cur) {
        if (n == null) return 0;
        
        cur = cur * 10 + n.val;
        
        if (n.left == null && n.right == null) {
            return cur;
        }
        
        return dfs(n.left, cur) + dfs(n.right, cur);
    }
}

File: problems/Python/129. Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        def dfs(n, cur):
            if not n:
                return 0
            
            cur = cur * 10 + n.val
            
            if not n.left and not n.right:
                return cur
            
            return dfs(n.left, cur) + dfs(n.right, cur)
            
        return dfs(root, 0)